docs/design/cloud-init.rst

```diff
@@ -0,0 +1,260 @@
+``mounts`` key of the ``cc_mounts`` module:
+```
+mounts:
+    - [ /dev/ephemeral0, /mnt, auto, "defaults,noexec" ]
+    - [ sdc, /opt/data ]
+    - [ xvdh, /opt/data, "auto", "defaults,nofail", "0", "0" ]
+```
+With this module we could create the swap entry for the disk. The problem is
+that we are using non ephemeral device names (UUID for swap) and this module
+does not support this. In order to know the correct device naming we should
+forward the ``DISK_%N_FRONTEND_TYPE`` from *Ganeti* to *snf-image-helper* which
+we don't. For now, its better if we bypass cloud-init and put the swap entry
+directly to fstab, if the swap is a whole disk.
+
+
+**50AssignHostname**: Adding a new ``local-hostname`` key in the metadata file
+should be enough to set the hostname. Alternatively, we could make use of the
+``cc_update_hostname`` module which supports the following keys:
+```
+preserve_hostname: <true/false>
+fqdn: <fqdn>
+hostname: <fqdn/hostname>
+```
+We could ignore the fqdn key and use the other two.
+
+**50ChangeMachineId**: We should probably leave this intact. Newer cloud-init
+version will automatically change the MachineId to a random value as this task
+does, but leaving it to work with older versions of cloud-init does not harm.
+
+**50ChangePassword**: This task should make use of the ``cc_set_password``
+module:
+```
+   ssh_pwauth: <yes/no/unchanged>
+
+    password: password1
+    chpasswd:
+        expire: <true/false>
+
+    chpasswd:
+        list: |
+            user1:password1
+            user2:RANDOM
+            user3:password3
+            user4:R
+```
+
+The ``password`` key only works for the default user and is not present in
+older versions of cloud-init. In order to make cloud-init work with the
+``USERS`` list we provide, chpasswd should be used. This task also puts ssh
+authorization keys to the user's home directories. To do this we need to set
+the ``public-keys`` meta-data key or use the ``ssh_authorized_keys`` config key
+of ``cc_ssh`` module. The problem is that this will put the keys only to the
+default user and root. You may put ssh keys to other users only when defining
+them using the ``users`` list of the ``cc_users_group`` module:
+
+```
+users:
+    - default
+    - name: <username>
+      expiredate: <date>
+      gecos: <comment>
+      groups: <additional groups>
+      homedir: <home directory>
+      inactive: <true/false>
+      lock_passwd: <true/false>
+      no-create-home: <true/false>
+      no-log-init: <true/false>
+      no-user-group: <true/false>
+      passwd: <password>
+      primary-group: <primary group>
+      selinux-user: <selinux username>
+      shell: <shell path>
+      snapuser: <email>
+      ssh-authorized-keys:
+          - <key>
+          - <key>
+      ssh-import-id: <id>
+      sudo: <sudo config>
+      system: <true/false>
+      uid: <user id>
+```
+
+but we don't want snf-image to ovewrite the already defined users or
+read/update existing configuration files.
+
```

Comment:
I am not sure the design doc is clear on what is supposed to happen. Currently there is more focus in this paragraph on what were the problems restriction, can we clarify what is supposed to happen at the end ?